agent/structs/structs.go

```diff
@@ -1153,16 +1154,16 @@ func (c *IndexedConfigEntries) MarshalBinary() (data []byte, err error) {
 	bs := make([]byte, 128)
 	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
 
-	// Encode kinds of entries first
+	// Encode length.
 	err = enc.Encode(len(c.Entries))
 	if err != nil {
 		return nil, err
 	}
```

Comment:
We shouldn't need to encode length at all any more right - msgpack will do that as part of the slice serialisation and we only have a single kind value before the main struct now.

We can just decode into a nil `IndexedConfigEntries` and msgpack will figure out length and allocations etc.

Reply:
I think we still need to fill in a dummy entry for each spot in the list with `MakeConfigEntry` - I couldn't find a way for it to work without populating the slice beforehand.